Extensions/vanilla_video.js

```diff
@@ -1,45 +1,115 @@
+				"25%", 0.25
+			],
+		},
+		"pause_on_click": {
+			text: "Play/pause the video by clicking on it",
+			default: true,
+			value: true
+		},
+		"loop": {
+			text: "Loop by default",
+			default: false,
+			value: false
+		},
+		"use_hd": {
+			text: "Use HD videos where available",
+			default: false,
+			value: false
+		},
+		"background_color": {
+			text: "Background color (<span style=\"text-decoration: underline; cursor: pointer;\" id=\"xkit-vanilla-video-color-help\">what is this?</span>)",
+			type: "text",
+			default: "rgba(0, 0, 0, 1.0)",
+			value: "rgba(0, 0, 0, 1.0)"
+		},
+		"disable_preload": {
+			text: "Disable video preloading (may save data)",
+			default: false,
+			value: false
+		},
+	},
+
 	running: false,
 	run: function() {
 		this.running = true;
+		if (this.preferences.background_color.value === "") {
+			this.preferences.background_color.value = "rgba(0, 0, 0, 1.0)";
+		}
 		XKit.post_listener.add('vanilla_video', XKit.extensions.vanilla_video.check);
 		XKit.extensions.vanilla_video.check();
 	},
+	
+	cpanel: function() {
+		$("#xkit-vanilla-video-color-help").click(function() {
+			XKit.window.show("Background color", "The Vanilla Videos extension allows you to set the background color used when videos do not fill the whole player, generally due to being very narrow. <br/><br/>Please be careful while customizing the color. An improper value can cause issues. In that case, just delete the text you've entered completely and XKit will revert to the default color.", "info", "<div class=\"xkit-button default\" id=\"xkit-close-message\">OK</div>");
+		});
+	},
 
 	check: function() {
 		var doneClass = 'xvanilla_video-done';
 		$('.posts .post').not('.' + doneClass).each(function() {
 			var post = $(this);
 			post.addClass(doneClass);
-			var videoEmbeds = post.find('.dockable_video_embed');
+			var videoEmbeds = post.find('.dockable_video_embed').add('.crt-video');
 			videoEmbeds.each(function() {
 				var videoEmbed = $(this);
-				if (videoEmbed.attr('data-embed-service') !== 'tumblr_video') {
+				if (videoEmbed.attr('data-embed-service') !== 'tumblr_video' && !videoEmbed.hasClass('crt-video')) {
 					return;
 				}
 				var sources = videoEmbed.find('video > source');
 				var newVideo = document.createElement('video');
 				sources.each(function() {
+					var origVideo = this.parentElement;
 					var clonedSource = document.createElement('source');
 					clonedSource.type = this.type;
 					clonedSource.src = this.src;
+					if (XKit.extensions.vanilla_video.preferences.use_hd.value) {
+						clonedSource.src = clonedSource.src.replace(/\/480$/, "");
+					}
+					newVideo.poster = origVideo.poster;
 					newVideo.appendChild(clonedSource);
 				});
+				if ($(newVideo).find("source").length === 0) {
+					// remove doneClass?
+					return; // FIXME sometimes a page ends up with all broken videos (missing source) - not sure why. haven't seen it since I added this but might just be a coincidence
```

Comment:
No; those same videos if liked would then play fine in my likes page with no change in options for Vanilla Videos. Or if seen again would play fine.